pkg/dbutil/index.go

```diff
@@ -113,6 +114,49 @@ func FindSuitableIndex(ctx context.Context, db *sql.DB, schemaName string, table
 	return c, nil
 }
 
+// FindAllIndex returns all index, order is pk, uk, and normal index.
+func FindAllIndex(tableInfo *model.TableInfo) []*model.IndexInfo {
+	indices := make([]*model.IndexInfo, len(tableInfo.Indices))
+	copy(indices, tableInfo.Indices)
+	sort.SliceStable(indices, func(i, j int) bool {
+		a := indices[i]
+		b := indices[j]
+		switch {
+		case b.Primary:
+			return false
+		case a.Primary:
+			return true
+		case b.Unique:
+			return false
+		case a.Unique:
+			return true
+		default:
+			return false
+		}
+	})
+	return indices
+}
+
+// FindAllColumnWithIndex returns columns with index, order is pk, uk and normal index.
+func FindAllColumnWithIndex(tableInfo *model.TableInfo) []*model.ColumnInfo {
+	colsMap := make(map[string]interface{})
+	cols := make([]*model.ColumnInfo, 0, 2)
+
+	for _, index := range FindAllIndex(tableInfo) {
+		// index will be guaranteed to be visited in order PK -> UK -> IK
+		for _, indexCol := range index.Columns {
+			col := FindColumnByName(tableInfo.Columns, indexCol.Name.O)
+			if _, ok := colsMap[col.Name.O]; ok {
+				continue
+			}
+			colsMap[col.Name.O] = struct{}{}
+			cols = append(cols, col)
```

Comment:
if we have two UK `ab` and `ac`, the returned `cols` may `abc` or `acb`, depends on their order in `tableInfo`, right?

Reply:
yes, it dosen't matter